src/main/java/org/cancogenvirusseq/muse/components/PayloadFileMapper.java

```diff
@@ -142,16 +144,16 @@ public PayloadFileMapper(String payloadJsonTemplate) {
 
   @SneakyThrows
   private static ObjectNode convertRecordToPayload(
-      Map<String, String> valuesMap, String payloadTemplate) {
+      Map<String, Object> valuesMap, String payloadTemplate) {
     StringLookup lookupFunc =
         key -> {
           val value = valuesMap.getOrDefault(key, "");
 
           // value is mapped to json value by these rules
-          if (NumberUtils.isCreatable(value)) {
+          if (value instanceof Number) {
```

Comment:
this is the main crux of the bug. Changed the records to be Map<String, Object> and here we check if the type of Object is Number